src/models/blockchain/block/block_body.rs

```diff
@@ -78,6 +123,10 @@ impl MastHash for BlockBody {
             self.block_mmr_accumulator.encode(),
         ]
     }
+
+    fn mast_hash(&self) -> Digest {
+        *self.mast_hash.get_or_init(|| self.merkle_tree().root())
```

Comment:
I'm not sure I get how this is safer. If the struct is not mutable outside this module, what can go wrong with the current solution?

Reply:
nothing to do with mutating the struct. 

This is about ensuring the cached data is used ie that no callers are using inefficient methods.  The surest way to prevent callers from doing that is to make such usage impossible.

Caching the digest:
1. callers can call merkle_tree().root() directly to obtain a hash, which bypasses the cache.  
2. they can also call mast_path() which invokes merkle_tree(), or merkle_tree() directly.  The default impl of merkle_tree() performs a bunch of hash ops.

if we impl merkle_tree() and cache its result then no caller can bypass that.

Reply:
Ah! So it's about efficiency. Got it.